lib/xcodeproj/project/object/native_target.rb

```diff
@@ -416,6 +416,19 @@ class PBXNativeTarget < AbstractTarget
         # @!group Helpers
         #--------------------------------------#
 
+        # Checks product_type to determine whether or not the recevier is a
+        # an app extension
+        #
+        # @return True if the target would be an extension embedded in an
+        #         app target, otherwise false
+        #
+        # @note watchOS 2 extensions are extensions of a watch target, not
+        #       an app target, whereas this is not the case for watchOS 1
+        def app_extension?
```

Comment:
Oh I see there's `extension_target_type?`, which I either didn't see or was newer than when I initially started this branch. That said, `app_extension?` specifically doesn't consider watch os 2 targets (unsure about tv targets), since the intent is to use this flag to determine if it's okay to embed frameworks in the target, which it's not for "app" extensions but I believe is okay to do for watch os 2 targets (cc @neonichu). Maybe I just need a better name here. How about `target_allows_embed_frameworks?` (and then I'll invert the condition)?